libnavigation-metrics/proguard-rules.pro

```diff
@@ -1,21 +1 @@
-# Add project specific ProGuard rules here.
-# You can control the set of applied configuration files using the
-# proguardFiles setting in build.gradle.
-#
-# For more details, see
-#   http://developer.android.com/guide/developing/tools/proguard.html
-
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
-
-# Uncomment this to preserve the line number information for
-# debugging stack traces.
-#-keepattributes SourceFile,LineNumberTable
-
-# If you keep the line number information, uncomment this to
-# hide the original source file name.
-#-renamesourcefileattribute SourceFile
+-keep class com.mapbox.navigation.metrics.internal.** { *; }
```

Comment:
Instead of keeping the whole `internal` package we should only keep `NavigationAppUserTurnstileEvent`

Reply:
made some changes: moved `NavigationAppUserTurnstileEvent` to `om.mapbox.navigation.metrics.internal.event` package and updated rule,
 moved `MetricEx` to  `com.mapbox.navigation.metrics.extensions` package and renamed  to`MetricEventEx`(no need internal package, it has `internal` access modifier)